modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-inventory-impl/src/main/java/com/liferay/headless/commerce/admin/inventory/internal/resource/v1_0/ReplenishmentItemResourceImpl.java

```diff
@@ -0,0 +1,240 @@
+)
+public class ReplenishmentItemResourceImpl
+	extends BaseReplenishmentItemResourceImpl {
+
+	@Override
+	public void deleteReplenishmentItem(Long replenishmentItemId)
+		throws Exception {
+
+		_commerceInventoryReplenishmentItemService.
+			deleteCommerceInventoryReplenishmentItem(replenishmentItemId);
+	}
+
+	@Override
+	public void deleteReplenishmentItemByExternalReferenceCode(
+			String externalReferenceCode)
+		throws Exception {
+
+		_commerceInventoryReplenishmentItemService.
+			deleteCommerceInventoryReplenishmentItem(
+				externalReferenceCode, contextCompany.getCompanyId());
+	}
+
+	@Override
+	public ReplenishmentItem getReplenishmentItem(Long replenishmentItemId)
+		throws Exception {
+
+		return _toReplenishmentItem(
+			_commerceInventoryReplenishmentItemService.
+				getCommerceInventoryReplenishmentItem(replenishmentItemId));
+	}
+
+	@Override
+	public ReplenishmentItem getReplenishmentItemByExternalReferenceCode(
+			String externalReferenceCode)
+		throws Exception {
+
+		return _toReplenishmentItem(
+			_getReplenishmentItemByExternalReferenceCode(
+				externalReferenceCode));
+	}
+
+	@Override
+	public Page<ReplenishmentItem> getReplenishmentItemsPage(
+			String sku, Pagination pagination)
+		throws Exception {
+
+		return Page.of(
+			TransformUtil.transform(
+				_commerceInventoryReplenishmentItemService.
+					getCommerceInventoryReplenishmentItemsByCompanyIdAndSku(
+						contextCompany.getCompanyId(), sku,
+						pagination.getStartPosition(),
+						pagination.getEndPosition()),
+				this::_toReplenishmentItem),
+			pagination,
+			_commerceInventoryReplenishmentItemService.
+				getCommerceInventoryReplenishmentItemsCountByCompanyIdAndSku(
+					contextCompany.getCompanyId(), sku));
+	}
+
+	@Override
+	public Page<ReplenishmentItem> getWarehouseIdReplenishmentItemsPage(
+			Long warehouseId, Pagination pagination)
+		throws Exception {
+
+		return Page.of(
+			TransformUtil.transform(
+				_commerceInventoryReplenishmentItemService.
+					getCommerceInventoryReplenishmentItemsByCommerceInventoryWarehouseId(
+						warehouseId, pagination.getStartPosition(),
+						pagination.getEndPosition()),
+				this::_toReplenishmentItem),
+			pagination,
+			_commerceInventoryReplenishmentItemService.
+				getCommerceInventoryReplenishmentItemsCountByCommerceInventoryWarehouseId(
+					warehouseId));
+	}
+
+	@Override
+	public Response patchReplenishmentItem(
+			Long replenishmentItemId, ReplenishmentItem replenishmentItem)
+		throws Exception {
+
+		CommerceInventoryReplenishmentItem commerceInventoryReplenishmentItem =
+			_commerceInventoryReplenishmentItemService.
+				getCommerceInventoryReplenishmentItem(replenishmentItemId);
+
+		_updateReplenishmentItem(
+			replenishmentItem, commerceInventoryReplenishmentItem);
+
+		Response.ResponseBuilder responseBuilder = Response.noContent();
+
+		return responseBuilder.build();
+	}
+
+	@Override
+	public Response patchReplenishmentItemByExternalReferenceCode(
```

Comment:
why PATCH is not returning the whole object?

Reply:
I had followed what is currently done for warehouses, I changed the method